modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/communicators/aah_async_comm.cc

```diff
@@ -70,192 +62,157 @@ AAH_ASynchronousCommunicator::Reset()
   data_initialized_ = false;
   upstream_data_initialized_ = false;
 
-  for (auto& message_flags : prelocI_message_received_)
-    message_flags.assign(message_flags.size(), false);
+  for (auto& rcv_flags : preloc_msg_received_)
+    rcv_flags.assign(rcv_flags.size(), false);
 
-  for (auto& message_flags : delayed_prelocI_message_received_)
-    message_flags.assign(message_flags.size(), false);
+  for (auto& rcv_flags : delayed_preloc_msg_received_)
+    rcv_flags.assign(rcv_flags.size(), false);
 }
 
 void
 AAH_ASynchronousCommunicator::BuildMessageStructure()
 {
   const auto& spds = fluds_.GetSPDS();
-  auto& fluds = dynamic_cast<AAH_FLUDS&>(fluds_);
-
-  // Predecessor locations
-  size_t num_dependencies = spds.GetLocationDependencies().size();
-
-  prelocI_message_count_.resize(num_dependencies, 0);
-  prelocI_message_size_.resize(num_dependencies);
-  prelocI_message_blockpos_.resize(num_dependencies);
-  prelocI_message_received_.clear();
+  const auto& fluds = dynamic_cast<AAH_FLUDS&>(fluds_);
 
-  for (int prelocI = 0; prelocI < num_dependencies; ++prelocI)
+  auto message_count_and_size = [this](const auto num_unknowns)
   {
-    size_t num_unknowns = fluds.GetPrelocIFaceDOFCount(prelocI) * num_groups_ * num_angles_;
     size_t message_count = num_angles_;
     if (num_unknowns * 8 > max_mpi_message_size_)
       message_count = ((num_unknowns * 8) + (max_mpi_message_size_ - 1)) / max_mpi_message_size_;
     size_t message_size = (num_unknowns + (message_count - 1)) / message_count;
+    return std::make_pair(message_count, message_size);
+  };
 
-    prelocI_message_count_[prelocI] = message_count;
-    prelocI_message_size_[prelocI].reserve(message_count);
-    prelocI_message_blockpos_[prelocI].reserve(message_count);
+  // Predecessor locations
+  const size_t num_dependencies = spds.GetLocationDependencies().size();
+  preloc_msg_data_.resize(num_dependencies);
+  preloc_msg_received_.resize(num_dependencies);
+
+  for (auto i = 0; i < num_dependencies; ++i)
+  {
+    size_t num_unknowns = fluds.GetPrelocIFaceDOFCount(i) * num_groups_ * num_angles_;
+    auto [message_count, message_size] = message_count_and_size(num_unknowns);
+    const auto locJ = spds.GetLocationDependencies()[i];
```

Comment:
Good to see all of the capitals in variables go away. Here is one that was missed. I would recommend `loc` or `deploc` here.

Reply:
Nice catch! Fixed.